qemu/domain_test.go

```diff
@@ -581,22 +564,44 @@ func TestEvents(t *testing.T) {
 // Test when a listener connects, but disconnects without
 // receiving from the events channel returned.
 func TestEventsDerelictListener(t *testing.T) {
-	d, done := testDomain(t, nil)
-	defer done()
+	mon := &testMonitor{}
 
-	_, stop, err := d.Events()
+	d1, err := NewDomain(mon, "test")
 	if err != nil {
 		t.Error(err)
 	}
+	defer d1.Close()
 
-	<-time.After(200 * time.Millisecond)
+	d2, err := NewDomain(mon, "test")
+	if err != nil {
+		t.Error(err)
+	}
+	defer d2.Close()
 
-	stop <- struct{}{}
-	select {
-	case <-stop:
-		return
-	case <-time.After(time.Millisecond * 20):
-		t.Error("shutdown failed to complete")
+	_, stop, err := d1.Events()
+	if err != nil {
+		t.Error(err)
+	}
+
+	events2, stop2, err := d2.Events()
+	if err != nil {
+		t.Error(err)
+	}
+
+	time.Sleep(200 * time.Millisecond)
+	close(stop)
+	close(stop2)
+
+	for {
+		select {
+		case _, ok := <-events2:
+			if !ok {
+				return
+			}
+		case <-time.After(1 * time.Second):
+			t.Error("shutdown failed to complete")
+			return
+		}
 	}
```

Comment:
Why are there two test domains?

Also, what do you think of pushing the timeout up into the `go test` runner so that it is configurable by whoever is running `go test` and we have one less reference to the `time` package in the test suite:

```suggestion
	for _ = range events2 {
	}
```

Reply:
> Why are there two test domains?

Oops, there shouldn't be. There should just be 2 event streams.

> Also, what do you think of pushing the timeout up into the go test runner so that it is configurable by whoever is running go test and we have one less reference to the time package in the test suite:

Sounds like a good idea to me!